### DISCUSSION TITLE
Capture events from ethers

### DISCUSSION THREAD

**Discussion by Makabongwe-M** [NomicFoundation/hardhat#5830] [Troubleshooting] [answered]
Hi there!

I am trying to capture events emitted by Solidity from the ethers side of the code. I have an even emitter `myEventEmitter(string _variable, uint32 _id)` in the Solidity contract. I then run a script that interacts with the function that invokes this event, I now want to grab this even and do things with it, but I don't get the event with the parameters I sent to the contract. I tried this [this](https://github.com/ethers-io/ethers.js/discussions/4484#discussioncomment-9890653) but I don't get the parameters. Please help.

**Comment by kanej** [comment, marked as answer]
This is the Hardhat project, this sounds like a more general Solidity or Ethers issue.
The [ethers docs](https://docs.ethers.org/v6/getting-started/) or the [Ethereum Stackexchange](https://ethereum.stackexchange.com/questions/152652/how-to-access-the-event-args-while-testing-using-ethers-v6) are likely better places to answer this question.
